### ISSUE TITLE
Confllict with `siunitx`

### ISSUE THREAD

**Issue by Stone-Zeng** [wspr/fontspec#301]
The following codes will give an error:
```tex
\documentclass{article}
\usepackage{fontspec}
\usepackage{siunitx}

\begin{document}
  Hello, world! \SI{4}{\centi\metre}
\end{document}
```

The error messages are
```tex
! Undefined control sequence.
<argument> \c_sys_engine_xetex_bool 
                                    
l.5 \begin{document}
```

XeLaTeX and LuaLaTeX will give the same result.

If I don't use `fontspec`, then it will not happen.

I use `fontspec` v2.6c (update just now) and `siunitx` v2.7h on TeX Live 2017.

**Comment by Stone-Zeng** [comment]
See issue [#311](https://github.com/josephwright/siunitx/issues/311) of `siunitx`.

**Comment by tweh** [comment]
This should be solved with the current `siunitx` version. At least I can’t reproduce the error on my system.

**Comment by wspr** [comment]
Appears to be fixed with siunitx 2.7i.
